olive/cache.py

```diff
@@ -9,82 +9,103 @@
 from pathlib import Path
 from typing import Dict, Optional, Union
 
-from olive.common.config_utils import serialize_to_json
-from olive.common.utils import hash_dict
-from olive.resource_path import ResourcePath, create_resource_path
+from olive.common.config_utils import ConfigBase, convert_configs_to_dicts, serialize_to_json, validate_config
+from olive.common.utils import hash_dict, set_nested_dict_value
+from olive.resource_path import ResourcePath, create_resource_path, find_all_resources
 
 logger = logging.getLogger(__name__)
 
 
-def get_cache_sub_dirs(cache_dir: Union[str, Path] = ".olive-cache"):
-    """Return the subdirectories of the cache directory.
+def set_cache_dir(cache_dir: Union[str, Path]):
+    """Set the cache directory."""
+    os.environ["OLIVE_CACHE_DIR"] = str(cache_dir)
 
-    There are three subdirectories: models, runs, and evaluations.
+
+def unset_cache_dir():
+    """Unset the cache directory."""
+    if "OLIVE_CACHE_DIR" in os.environ:
+        del os.environ["OLIVE_CACHE_DIR"]
+
+
+def get_cache_dir() -> Path:
+    """Return the cache directory.
+
+    The cache directory is set by the OLIVE_CACHE_DIR environment variable.
+    If the environment variable is not set, the default cache directory is .olive-cache.
     """
-    cache_dir = Path(cache_dir)
-    return cache_dir / "models", cache_dir / "runs", cache_dir / "evaluations", cache_dir / "non_local_resources"
+    return Path(os.environ.get("OLIVE_CACHE_DIR", ".olive-cache")).resolve()
 
 
-def clean_cache(cache_dir: Union[str, Path] = ".olive-cache"):
+def get_cache_sub_dirs() -> Dict[str, Path]:
+    """Return the subdirectories of the cache directory."""
+    cache_dir = get_cache_dir()
+    return {
+        "models": cache_dir / "models",
```

Comment:
Maybe you missed `workflow_id` which is added by Xiaoyu.
![image](https://github.com/user-attachments/assets/9573b2a4-f4d1-4e02-bce0-e9870ea87a10)

Reply:
This is set during engine init as cache_dir attribute https://github.com/microsoft/Olive/blob/97df601aae7648f8482e8f8c819ac765350508ef/olive/engine/engine.py#L91

Reply:
But when you see `get_cache_dir`, it still use the default value of `.olive-cache`.
Actually, in olive when you call `get_cache_dir`, the `OLIVE_CACHE_DIR` must be set already, right?

Reply:
For the case where user set the cache dir by `OLIVE_CACHE_DIR`, then `get_cache_dir` will return the path without `workflow_id`.

Reply:
The environment variable is currently not expected to be set by the user but by the engine during initialization. The environment variable is only expected to be used by other parts of olive code to know where the cache dir is. 

However, this feature is not currently needed so I am removing it in my next commit. will re-introduce it when I need it for my hf model changes.